libmamba/src/core/subdirdata.cpp

```diff
@@ -826,7 +826,16 @@ namespace mamba
         {
             fs::u8path state_file = json_file;
             state_file.replace_extension(".state.json");
-            fs::rename(m_temp_file->path(), json_file);
```

Comment:
hmm, now we don't have `rename` at all? I think we should first try to do the rename, then the move. I'll update your PR.